File: ios/UremontWhoop/OBD/ObdStandardLabels.swift
import Foundation

/// Подписи стандартных PID Mode 01 (SAE J1979).
enum ObdStandardLabels {

    static func obdStandard1c(_ code: Int) -> String? {
        switch code {
        case 1: return "OBD-II (CARB)"
        case 2: return "OBD (EPA)"
        case 3: return "OBD + OBD-II"
        case 4: return "OBD-I"
        case 5: return "Без OBD"
        case 6: return "EOBD (Европа)"
        case 7: return "EOBD + OBD-II"
        case 8: return "EOBD + OBD"
        case 9: return "EOBD + OBD + OBD-II"
        case 10: return "JOBD (Япония)"
        case 11: return "JOBD + OBD-II"
        case 12: return "JOBD + EOBD"
        case 13: return "JOBD + EOBD + OBD-II"
        case 14: return "Индия (Bharat)"
        case 15: return "Индия + OBD-II"
        case 16: return "HD OBD (тягачи)"
        case 17: return "HD OBD + OBD-II-C"
        case 18: return "HD EOBD-I"
        case 19: return "HD EOBD-I N"
        case 20: return "HD EOBD-I + HD EOBD-II N"
        case 21: return "HD EOBD-II N"
        case 22: return "HD EOBD-II + HD EOBD-II N"
        case 23: return "WOBD-I"
        default: return (1...255).contains(code) ? "OBD (код \(code))" : nil
        }
    }

    static func fuelType51(_ code: Int) -> String? {
        switch code {
        case 0: return nil
        case 1: return "Бензин"
        case 2: return "Метанол"
        case 3: return "Этанол"
        case 4: return "Дизель"
        case 5: return "LPG"
        case 6: return "CNG"
        case 7: return "Пропан"
        case 8: return "Электричество"
        case 9: return "Бензин + газ (bi-fuel)"
        case 10: return "Бензин + метанол"
        case 11: return "Бензин + этанол"
        case 12: return "Бензин + электричество"
        case 13: return "Дизель + электричество"
        case 14: return "Гибрид (бензин/электро)"
        case 15: return "Гибрид (дизель/электро)"
        case 16: return "Гибрид (смешанный)"
        case 17: return "Гибрид (регенеративный)"
        case 18: return "Бензин + CNG"
        case 19: return "Бензин + LPG"
        case 20: return "Бензин + CNG + LPG"
        case 21: return "Гибрид (бензин + электро, внешняя зарядка)"
        case 22: return "Гибрид (дизель + электро, внешняя зарядка)"
        case 23: return "Гибрид (смешанный, внешняя зарядка)"
        default: return (1...255).contains(code) ? "Топливо (код \(code))" : nil
        }
    }
}

File: android/app/src/main/java/com/uremont/bluetooth/ObdStandardLabels.kt
package com.uremont.bluetooth

/**
 * Человекочитаемые подписи для стандартных PID Mode 01 (SAE J1979 / ISO 15031).
 */
object ObdStandardLabels {

    /** PID 0x1C — тип требований OBD / регион. */
    fun obdStandard1c(code: Int): String? = when (code) {
        1 -> "OBD-II (CARB)"
        2 -> "OBD (EPA)"
        3 -> "OBD + OBD-II"
        4 -> "OBD-I"
        5 -> "Без OBD"
        6 -> "EOBD (Европа)"
        7 -> "EOBD + OBD-II"
        8 -> "EOBD + OBD"
        9 -> "EOBD + OBD + OBD-II"
        10 -> "JOBD (Япония)"
        11 -> "JOBD + OBD-II"
        12 -> "JOBD + EOBD"
        13 -> "JOBD + EOBD + OBD-II"
        14 -> "Индия (Bharat)"
        15 -> "Индия + OBD-II"
        16 -> "HD OBD (тягачи)"
        17 -> "HD OBD + OBD-II-C"
        18 -> "HD EOBD-I"
        19 -> "HD EOBD-I N"
        20 -> "HD EOBD-I + HD EOBD-II N"
        21 -> "HD EOBD-II N"
        22 -> "HD EOBD-II + HD EOBD-II N"
        23 -> "WOBD-I"
        else -> if (code in 24..255) "OBD (код $code)" else null
    }

    /** PID 0x51 — тип топлива двигателя. */
    fun fuelType51(code: Int): String? = when (code) {
        0 -> null
        1 -> "Бензин"
        2 -> "Метанол"
        3 -> "Этанол"
        4 -> "Дизель"
        5 -> "LPG"
        6 -> "CNG"
        7 -> "Пропан"
        8 -> "Электричество"
        9 -> "Бензин + газ (би-fuel)"
        10 -> "Бензин + метанол"
        11 -> "Бензин + этанол"
        12 -> "Бензин + электричество"
        13 -> "Дизель + электричество"
        14 -> "Гибрид (бензин/электро)"
        15 -> "Гибрид (дизель/электро)"
        16 -> "Гибрид (смешанный)"
        17 -> "Гибрид (регенеративный)"
        18 -> "Бензин + CNG"
        19 -> "Бензин + LPG"
        20 -> "Бензин + CNG + LPG"
        21 -> "Гибрид (бензин + электро, внешняя зарядка)"
        22 -> "Гибрид (дизель + электро, внешняя зарядка)"
        23 -> "Гибрид (смешанный, внешняя зарядка)"
        else -> if (code in 1..255) "Топливо (код $code)" else null
    }
}
